recipes/hyperscan/all/patches/fix-cmake.patch

```diff
@@ -0,0 +1,251 @@
+diff --git CMakeLists.txt CMakeLists.txt
+index 8bc6077..60c2ffd 100644
+--- CMakeLists.txt
++++ CMakeLists.txt
+@@ -1,4 +1,4 @@
+-cmake_minimum_required (VERSION 2.8.11)
++cmake_minimum_required (VERSION 3.0.0)
+ project (hyperscan C CXX)
+ 
+ set (HS_MAJOR_VERSION 5)
+@@ -6,7 +6,7 @@ set (HS_MINOR_VERSION 4)
+ set (HS_PATCH_VERSION 0)
+ set (HS_VERSION ${HS_MAJOR_VERSION}.${HS_MINOR_VERSION}.${HS_PATCH_VERSION})
+ 
+-set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
++list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
+ include(CheckCCompilerFlag)
+ include(CheckCXXCompilerFlag)
+ include(CheckCXXSymbolExists)
+@@ -17,8 +17,8 @@ INCLUDE (CheckLibraryExists)
+ INCLUDE (CheckSymbolExists)
+ include (CMakeDependentOption)
+ include (GNUInstallDirs)
+-include (${CMAKE_MODULE_PATH}/platform.cmake)
+-include (${CMAKE_MODULE_PATH}/ragel.cmake)
++include (platform)
++include (ragel)
+ 
+ find_package(PkgConfig QUIET)
+ 
+@@ -69,7 +69,7 @@ include_directories(${PROJECT_SOURCE_DIR}/src)
+ include_directories(${PROJECT_BINARY_DIR})
+ include_directories(SYSTEM include)
+ 
+-include (${CMAKE_MODULE_PATH}/boost.cmake)
++include (boost)
```

Comment:
I suspected these do not find the conan generated targets 🤔 

Can you explain these changes please

Reply:
Could you explain or point me to the respective documentation about "conan generated targets"? Following the CMake documentation for `include()`, the file extension has not to be given when the modules are on the module path (which they are, because the patch appends the "cmake" subdirectory to the module path.

Reply:
So conan works by converting the `cpp_info` into build/project system files that can be included.

making the build files is done by https://docs.conan.io/en/latest/reference/generators.html

You have opted for the https://github.com/conan-io/conan-center-index/blob/a72ce585e27325b8b014c3eca26a1605dc2f3ded/recipes/hyperscan/all/conanfile.py#L14

You can scan through the documentation https://docs.conan.io/en/latest/reference/generators/cmake.html but everything has a conan prefix...

Right now the project is using it's own custom cmake scripts which do no take Conan into account so the dependencies are (mayber) not being used.

You might want to add something to the test package that calls something that requires boost so we can see if it's linked correctly.

This way we can know for sure if addition changes are required.

Reply:
So what I did was a) Add a `message(STATUS "Boost libs = ${CONAN_LIB_DIRS_BOOST}")` to the upstream CMakeLists.txt and watch the output during `conan create`. The correct directory is printed (which was something like /home/user/.conan/data/boost/...), and b) Use boost program_options in the text package, which compiles and links fine. Note that I don't have boost libs installed system-wide which could have been picked up. So I'm pretty sure the changes work as expected.
I could push those little changes for review as well, although they are meant for debugging. I didn't want to stress the CI with those for now.

Reply:
It must be some magic happening in the cmake which I have never seen!

It works 🤷 LGTM!